catalogd/internal/source/containers_image.go

```diff
@@ -309,15 +315,21 @@ func (i *ContainersImageRegistry) unpackImage(ctx context.Context, unpackPath st
 			}
 			defer layerReader.Close()
 
-			if err := applyLayer(ctx, unpackPath, dirToUnpack, layerReader); err != nil {
+			if err := applyLayer(ctx, tempUnpackPath, dirToUnpack, layerReader); err != nil {
 				return fmt.Errorf("error applying layer[%d]: %w", i, err)
 			}
 			l.Info("applied layer", "layer", i)
 			return nil
 		}(); err != nil {
-			return errors.Join(err, deleteRecursive(unpackPath))
+			return errors.Join(err, deleteRecursive(tempUnpackPath))
 		}
 	}
+	if err := os.MkdirAll(unpackPath, 0700); err != nil {
+		return fmt.Errorf("error creating unpack directory: %w", err)
+	}
+	if err := os.Rename(tempUnpackPath, unpackPath); err != nil {
```

Comment:
I think there are some limitations with `Rename` renaming across filesystems (not sure if on the Pod /tmp and /var are different fs's though)